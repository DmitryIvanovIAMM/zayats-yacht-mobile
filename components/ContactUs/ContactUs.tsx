import { secondary } from "@/constants/Colors";
import { formatPhoneNumber } from "@/helpers/formatPhoneNumber";
import { MaterialIcons } from "@expo/vector-icons";
import {
  Linking,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import SectionTitle from "../SectionTitle/SectionTitle";
import { ThemedText } from "../ThemedText";

const PHONE_NUMBER = "+13051234567";

const ContactUs = () => {
  const handleEmailPress = () => {
    Linking.openURL("mailto:info@zayats-yacht.com");
  };

  const handlePhonePress = () => {
    Linking.openURL(`tel:${PHONE_NUMBER}`);
  };

  return (
    <View style={styles.container}>
      <View style={styles.innerContainer}>
        <View style={styles.title}>
          <SectionTitle>Contact Us</SectionTitle>
        </View>
        <View style={styles.item}>
          <View style={styles.iconWrapper}>
            <MaterialIcons
              name="location-on"
              size={32}
              color={secondary.dark}
            />
          </View>
          <Text style={styles.text}>
            <ThemedText style={styles.bold}>Address</ThemedText>
            {"\n"}
            Allied Yacht Transport, LLC
            {"\n"}
            12555 Orange Drive, Suite 107
            {"\n"}Fort Lauderdale, FL, 33330, USA
          </Text>
        </View>
        <View style={styles.item}>
          <View style={styles.iconWrapper}>
            <MaterialIcons name="email" size={32} color={secondary.dark} />
          </View>
          <Text style={styles.text}>
            <ThemedText style={styles.bold}>Email</ThemedText>
            {"\n"}
            <TouchableOpacity
              onPress={handleEmailPress}
              accessibilityRole="button"
              accessibilityLabel="Send email to info@zayats-yacht.com"
            >
              <Text style={[styles.link]}>info@zayats-yacht.com</Text>
            </TouchableOpacity>
          </Text>
        </View>
        <View style={styles.item}>
          <View style={styles.iconWrapper}>
            <MaterialIcons name="phone" size={32} color={secondary.dark} />
          </View>
          <Text style={styles.text}>
            <ThemedText style={styles.bold}>Phone</ThemedText>
            {"\n"}
            <TouchableOpacity
              onPress={handlePhonePress}
              accessibilityRole="button"
              accessibilityLabel="Call phone number +1 (305) 123-4567"
            >
              <Text style={styles.link}>{formatPhoneNumber(PHONE_NUMBER)}</Text>
            </TouchableOpacity>
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#F7F8F9",
    flexDirection: "column",
    marginBottom: 50,
    alignItems: "center",
  },
  innerContainer: {
    width: "90%",
    alignSelf: "center",
    marginBottom: 20,
  },
  title: {
    justifyContent: "center",
    alignItems: "center",
  },
  item: {
    width: "80%",
    flexDirection: "row",
    backgroundColor: "#F7F8F9",
    justifyContent: "flex-start",
    alignItems: "center",
    borderRadius: 8,
  },
  iconWrapper: {
    width: 40,
    height: 40,
    borderRadius: 30,
    backgroundColor: "#fff",
    justifyContent: "center",
    alignItems: "center",
    borderColor: "#e0e0e0",
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 0.08,
    shadowRadius: 2,
    elevation: 2,
    margin: 10,
    boxShadow: "0px 2px 20px rgba(51, 101, 167, 0.18)",
  },
  text: {
    fontSize: 16,
    color: secondary.dark,
    marginLeft: 12,
  },
  bold: {
    fontWeight: "bold",
    color: secondary.dark,
  },
  link: {
    textDecorationLine: "none",
    color: secondary.dark,
  },
});

export default ContactUs;
