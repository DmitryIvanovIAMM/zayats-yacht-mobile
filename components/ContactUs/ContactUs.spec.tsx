import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import { Linking } from "react-native";
import ContactUs from "./ContactUs";

// Mocks to avoid native/complex deps
jest.mock("@/components/SectionTitle/SectionTitle", () => {
  const React = require("react");
  const { Text } = require("react-native");
  return {
    __esModule: true,
    default: ({ children }: any) => <Text>{children}</Text>
  };
});

jest.mock("../ThemedText", () => {
  const React = require("react");
  const { Text } = require("react-native");
  return {
    __esModule: true,
    ThemedText: ({ children, style }: any) => (
      <Text style={style}>{children}</Text>
    )
  };
});

jest.mock("@expo/vector-icons", () => {
  const React = require("react");
  const { View } = require("react-native");
  return {
    MaterialIcons: (props: any) => <View accessibilityRole="image" {...props} />
  };
});

describe("ContactUs", () => {
  beforeEach(() => {
    jest.spyOn(Linking, "openURL").mockResolvedValue(undefined as any);
  });

  afterEach(() => {
    jest.restoreAllMocks();
    jest.clearAllMocks();
  });

  it("matches snapshot", () => {
    const ref = React.createRef<any>();
    const { toJSON } = render(<ContactUs ref={ref} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders title and contact details", () => {
    const ref = React.createRef<any>();
    const { getByText } = render(<ContactUs ref={ref} />);

    expect(getByText("Contact Us")).toBeTruthy();
    expect(getByText("Address")).toBeTruthy();
    expect(getByText("Email")).toBeTruthy();
    expect(getByText("Phone")).toBeTruthy();

    // Company and address lines
    expect(getByText(/Zayats Yacht Transport, LLC/i)).toBeTruthy();
    expect(getByText(/Fort Lauderdale/i)).toBeTruthy();

    // Email text
    expect(getByText(/info@zayats-yacht\.com/i)).toBeTruthy();
  });

  it("opens mail client when email is pressed", () => {
    const ref = React.createRef<any>();
    const { getByLabelText } = render(<ContactUs ref={ref} />);

    const emailButton = getByLabelText("Send email to info@zayats-yacht.com");
    fireEvent.press(emailButton);

    expect(Linking.openURL).toHaveBeenCalledWith(
      "mailto:info@zayats-yacht.com"
    );
  });

  it("opens dialer when phone is pressed", () => {
    const ref = React.createRef<any>();
    const { getByLabelText } = render(<ContactUs ref={ref} />);

    const phoneButton = getByLabelText("Call phone number +1 (555) 555-5555");
    fireEvent.press(phoneButton);

    expect(Linking.openURL).toHaveBeenCalledWith("tel:+15555555555");
  });
});
