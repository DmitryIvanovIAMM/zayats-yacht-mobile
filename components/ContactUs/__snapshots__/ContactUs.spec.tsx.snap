// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUs matches snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "center",
      "backgroundColor": "#F7F8F9",
      "flexDirection": "column",
    }
  }
  id="contact-us-section"
>
  <View>
    <View
      style={
        {
          "alignSelf": "center",
          "marginBottom": 20,
          "width": "90%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Text>
          Contact Us
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#F7F8F9",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "80%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderColor": "#e0e0e0",
              "borderRadius": 30,
              "boxShadow": "0px 2px 20px rgba(51, 101, 167, 0.18)",
              "elevation": 2,
              "height": 40,
              "justifyContent": "center",
              "margin": 10,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.08,
              "shadowRadius": 2,
              "width": 40,
            }
          }
        >
          <View
            accessibilityRole="image"
            color="#006666"
            name="location-on"
            size={32}
          />
        </View>
        <Text
          style={
            {
              "color": "#006666",
              "fontSize": 16,
              "marginLeft": 12,
            }
          }
        >
          <Text
            style={
              {
                "color": "#006666",
                "fontWeight": "bold",
              }
            }
          >
            Address
          </Text>
          

          Zayats Yacht Transport, LLC
          

          55555 Orange Drive, Suite 555
          

          Fort Lauderdale, FL, 33330, USA
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#F7F8F9",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "80%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderColor": "#e0e0e0",
              "borderRadius": 30,
              "boxShadow": "0px 2px 20px rgba(51, 101, 167, 0.18)",
              "elevation": 2,
              "height": 40,
              "justifyContent": "center",
              "margin": 10,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.08,
              "shadowRadius": 2,
              "width": 40,
            }
          }
        >
          <View
            accessibilityRole="image"
            color="#006666"
            name="email"
            size={32}
          />
        </View>
        <Text
          style={
            {
              "color": "#006666",
              "fontSize": 16,
              "marginLeft": 12,
            }
          }
        >
          <Text
            style={
              {
                "color": "#006666",
                "fontWeight": "bold",
              }
            }
          >
            Email
          </Text>
          

          <View
            accessibilityLabel="Send email to info@zayats-yacht.com"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#006666",
                    "textDecorationLine": "none",
                  },
                ]
              }
            >
              info@zayats-yacht.com
            </Text>
          </View>
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#F7F8F9",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "80%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderColor": "#e0e0e0",
              "borderRadius": 30,
              "boxShadow": "0px 2px 20px rgba(51, 101, 167, 0.18)",
              "elevation": 2,
              "height": 40,
              "justifyContent": "center",
              "margin": 10,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.08,
              "shadowRadius": 2,
              "width": 40,
            }
          }
        >
          <View
            accessibilityRole="image"
            color="#006666"
            name="phone"
            size={32}
          />
        </View>
        <Text
          style={
            {
              "color": "#006666",
              "fontSize": 16,
              "marginLeft": 12,
            }
          }
        >
          <Text
            style={
              {
                "color": "#006666",
                "fontWeight": "bold",
              }
            }
          >
            Phone
          </Text>
          

          <View
            accessibilityLabel="Call phone number +1 (555) 555-5555"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#006666",
                  "textDecorationLine": "none",
                }
              }
            >
              +1 (555) 555-5555
            </Text>
          </View>
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
